### ISSUE TITLE
C++ prototype

### ISSUE THREAD

**Issue by cnuke** [cbe-project/cbe#8]
For prototyping or rather for fast iterating design approaches, C++ is better suited than SPARK. My [cbe_wip](https://github.com/cnuke/genode/commits/cbe_wip) branch contains the WIP prototype and is updated regularly (I will fixup/squash commits at my choosing). It is mainly intended for working out the interaction between the various modules and their specific interfaces.

**Comment by nfeske** [comment]
Thanks a lot for sharing the working branch her.! By just skimming over it, I must say that it looks fantastic!

**Comment by cnuke** [comment]
The last couple of weeks I tinkered with a little tool that initializes the virtual block-device tree as well as the free-list tree. It started out as a POSIX CLI tool that traverses the trees recursively (after all the depth is bound by the height of the tree - it will be changed to an interative approach to ease the translation to SPARK though). I'm currently in the process of moving parts of it to a Genode component that uses a RAM disk as backing store and provides a Block service to its clients. Being RAM based has the advantage that all inner nodes are readily available but in case they are fetched from a Block device, that might not be the case. Therefor I'm thinking about introducing (or rather how it fits in the overall architecture) a cache table for the inner nodes that is used for the tree traversal. Eventually parts of the component will became modules for the CBE.

As I misremembered the original design of the CBE, the tool initially used a different strategy to manage the free blocks (the free-list basically covered all available blocks and the VBD was initialized sparsely), which I had to fix first.
